core/arch/riscv64/proc.c

```diff
@@ -140,8 +145,12 @@ proc_has_feature(feature_bit_t f)
 void
 machine_cache_sync(void *pc_start, void *pc_end, bool flush_icache)
 {
-    /* FIXME i#3544: Not implemented. FENCE.I ? Maybe a syscall to clean icache? */
-    ASSERT_NOT_IMPLEMENTED(false);
+    /* We need to flush the icache on all harts, which is not feasible for FENCE.I, so we
+     * use SYS_riscv_flush_icache to let the kernel do this.
+     * The flag here is set to SYS_RISCV_FLUSH_ICACHE_LOCAL, which is defined as 1 in the
```

Comment:
Let's make a #define (or const int) for SYS_RISCV_FLUSH_ICACHE_LOCAL and use it in the syscall here.

Reply:
It could just be local to this file for now; if we need it somewhere else we can move it to a header at that time.

Reply:
Done in https://github.com/DynamoRIO/dynamorio/pull/6209/commits/89b268e727be5e5d7d09167c2c3f0ea914ef5e31.